src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -191,7 +193,44 @@ public enum Keywords : long
             /// </summary>
             GCHeapSnapshot = GC | GCHeapCollect | GCHeapDump | GCHeapAndTypeNames | Type,
         };
-        public ClrTraceEventParser(TraceEventSource source) : base(source) { }
+        public ClrTraceEventParser(TraceEventSource source) : base(source)
+        {
+
+            // Subscribe to the GCBulkType events and remember the TypeID -> TypeName mapping. 
+            ClrTraceEventParserState state = State;
+            AddCallbackForEvents<GCBulkTypeTraceData>(delegate (GCBulkTypeTraceData data)
+            {
+                for (int i = 0; i < data.Count; i++)
+                {
+                    GCBulkTypeValues value = data.Values(i);
+                    string typeName = value.TypeName;
+                    // TODO the GCBulkType events are logged after the event that needed it.  It really
```

Comment:
:bulb: I wouldn't bother with the TODO here. What's done is done. A simple comment that we're ignoring the event time since they are reported after the event that uses them in some cases will suffice. (If you remove the `timeQPC` parameter of `SetTypeIDToName`, this comment could be moved inside that method.)